### PR TITLE
fix: check optional keyword before accessing

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*
#1910 

*Description of changes:*
Check optional Keyword `Environment`, `NetworkConfig` before accessing them in `ModelMonitor.attach`

*Testing done:*
`tox -e py36 -- tests/unit`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/monitor/test_model_monitoring.py [hunk 1]
```diff
@@ -153,6 +181,17 @@ def test_default_model_monitor_with_invalid_network_config(sagemaker_session):
     assert INTER_CONTAINER_ENCRYPTION_EXCEPTION_MSG in str(exception.value)
 
 
+def test_model_monitor_without_network_config(sagemaker_session):
+    my_model_monitor = ModelMonitor(
+        role=ROLE,
+        image_uri=CUSTOM_IMAGE_URI,
+        sagemaker_session=sagemaker_session,
+    )
+    model_monitor_schedule_name = "model-monitoring-without-netwotk-config"
```
Comment (ajaykarpur) [inline tests/unit/sagemaker/monitor/test_model_monitoring.py:190]:
```suggestion
    model_monitor_schedule_name = "model-monitoring-without-network-config"
```

File: src/sagemaker/model_monitor/model_monitoring.py [hunk 2]
```diff
@@ -672,15 +672,24 @@ def attach(cls, monitor_schedule_name, sagemaker_session=None):
...
         network_config_dict = schedule_desc["MonitoringScheduleConfig"][
             "MonitoringJobDefinition"
         ].get("NetworkConfig")
 
-        vpc_config = schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"][
-            "NetworkConfig"
-        ].get("VpcConfig")
+        vpc_config = None
+        if (
+            schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"].get(
```
Comment (icywang86rui) [inline src/sagemaker/model_monitor/model_monitoring.py:685]: should this be just ``if network_config_dict is not None``

Comment (icywang86rui) [reply src/sagemaker/model_monitor/model_monitoring.py:685]: I would also put all the network_config parsing stuff in the same if block.

**Comment by icywang86rui** [comment]
This attached function is over 100 lines long now. Should we move some of the dict parsing stuff out or maybe model the DescribeSchedule call output into a class and do all the parsing there?

File: src/sagemaker/model_monitor/model_monitoring.py (hunk 2 above)
Comment (ChuyangDeng) [reply src/sagemaker/model_monitor/model_monitoring.py]: That makes sense! I also refactored the `attach` method to make it more concise

**Review by ajaykarpur — APPROVED** [review]

**Pull request merged**